benchmarks/benchmarker.py

```diff
@@ -0,0 +1,329 @@
+                        fasters.append(info)
+                    if ratio == 1:
+                        print("\t{:6.6} and {:6.6} "
+                              "have the same speed".format(rt, ro))
+                        sames.append(info)
+
+                num_threads = len(this_res["thread_scale"]) - 1
+                for i in range(1, num_threads + 1):
+                    for j, N in enumerate(this_res["Ns"]):
+                        this_t = this_res["thread_scale"][i][j]
+                        other_t = other_res["thread_scale"][i][j]
+                        ratio = other_t/this_t
+
+                        print("Threads: {}, N: {}, "
+                              "ratio: {:0.2f}".format(str(i), N, ratio))
+
+                        info = {"name": this_res["name"],
+                                "params": this_res["params"],
+                                "threads": i,
+                                "N": N,
+                                "ratio": ratio}
+
+                        if ratio < 1:
+                            print("\t{:6.6} is {:0.2f} times "
+                                  "slower than {:6.6}".format(rt, ratio, ro))
+                            slowers.append(info)
+                        if ratio > 1:
+                            print("\t{:6.6} is {:0.2f} times "
+                                  "faster than {:6.6}".format(rt, ratio, ro))
+                            fasters.append(info)
+                        if ratio == 1:
+                            print("\t{:6.6} and {:6.6} "
+                                  "have the same speed".format(rt, ro))
+                            sames.append(info)
+
+                print('\n ----------------')
+
+    save_comparison_result(rt, ro, slowers, fasters, sames)
+
+    threshold = 0.70
+    fail = False
+    for info in slowers:
+        if info["ratio"] < threshold:
+            desc = benchmark_desc(info["name"], info["params"])
+            print("TOO SLOW (beyond threshold of {})".format(threshold))
+            print("\t" + desc)
+            print("\t\tratio = {}".format(info["ratio"], threshold))
+            fail = True
+    if fail:
+        sys.exit(1)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        "Test the runtime performance of freud")
+    subparsers = parser.add_subparsers()
+
+    parser_run = subparsers.add_parser(
+        name='run',
+        description="Execute performance tests in various categories for "
+                    "specific data space sizes (N).")
+    parser_run.add_argument(
+        '-o', '--output', nargs='?', default='benchmark.json',
+        help="Specify which collection file to store results "
+             "to or '-' for None, "
+             "default='benchmark.json'.")
+    parser_run.add_argument(
+        '-c', '--circleci', action='store_true',
+        help="Flag for running on circle.ci to fix thread number")
```

Comment:
I'd like to refactor this to take an integer and limit to that number of threads, instead of making it specific to CircleCI. The CircleCI config can specify 2 threads via the command line with something like `-j 2` or `--threads 2` instead of `-c`.

We could also remove the thread count from the signature of every `run` method, too -- there's probably a way we could use environment variables or similar, just to make the implementation of all this cleaner. Ask @vyasr for his opinion.